supportedBackends/spark/src/main/scala/cromwell/backend/impl/spark/SparkClusterProcess.scala

```diff
@@ -0,0 +1,165 @@
+  }
+
+}
+
+trait SparkClusterRestClient {
+  def sendAndReceive: SendReceive
+
+  def makeHttpRequest(httpRequest: HttpRequest): Future[HttpResponse]
+}
+
+trait SparkClusterProcessMonitor {
+  def startMonitoringSparkClusterJob(jobPath: Path, jsonFile: String): Future[TerminalStatus]
+
+  def monitorSparkClusterJob(subId: String, rcPath: Path, promise: Promise[Unit]): Unit
+
+  def evaluateMonitoringFuture(rcPath: Path): Unit
+
+  def completeMonitoringProcess(rcPath: Path, status: String, promise: Promise[Unit]): Unit
+}
+
+trait SparkClusterJobParser {
+  def parseJsonForSubmissionIdAndStatus(jsonFile: Path): SparkJobSubmissionResponse
+}
+
+class SparkClusterProcess(implicit system: ActorSystem) extends SparkProcess
+  with SparkClusterRestClient with SparkClusterJobParser with SparkClusterProcessMonitor {
+
+  import SparkClusterProcess._
+  import spray.httpx.SprayJsonSupport._
+  import SparkClusterJsonProtocol._
+
+  implicit lazy val ec: ExecutionContext = system.dispatcher
+  lazy val completionPromise = Promise[TerminalStatus]()
+  lazy val monitorPromise = Promise[Unit]()
+  val tag = this.getClass.getSimpleName
+  lazy val logger = Logger(LoggerFactory.getLogger(getClass.getName))
+
+  override def sendAndReceive: SendReceive = sendReceive
+
+  override def startMonitoringSparkClusterJob(jobPath: Path, jsonFile: String): Future[TerminalStatus] = {
+    Future(parseJsonForSubmissionIdAndStatus(jobPath.resolve(jsonFile))) onComplete {
+      case Success(resp: SparkJobSubmissionResponse) =>
+        val rcPath = jobPath.resolve("rc")
+        monitorSparkClusterJob(resp.submissionId, rcPath, monitorPromise)
+        evaluateMonitoringFuture(rcPath)
+      case Failure(exception: Throwable) =>
+        logger.error(s"{} Spark Job failed to submit successfully following reason: {}", tag, exception.getMessage)
+        completionPromise success Failed(exception)
+    }
+    completionPromise.future
+  }
+
+  override def monitorSparkClusterJob(subId: String, rcPath: Path, promise: Promise[Unit]): Unit = {
+    pollForJobStatus(subId) onComplete {
+      case Success(resp: SparkDriverStateQueryResponse) if resp.driverState == "RUNNING" =>
+        Thread.sleep(1000)
+        logger.debug(s"{} Spark Driver is now in :{} state", tag, "RUNNING")
+        monitorSparkClusterJob(subId, rcPath, promise)
+      case Success(resp: SparkDriverStateQueryResponse) if resp.driverState == "FINISHED" =>
+        logger.debug(s"{} Spark Driver is now in :{} state", tag, "FINISHED")
+        completeMonitoringProcess(rcPath, "0", promise)
+      case Success(resp: SparkDriverStateQueryResponse) if resp.driverState == "FAILED" =>
+        logger.debug(s"{} Spark Driver is now in :{} state", tag, "FAILED")
+        completeMonitoringProcess(rcPath, "-1", promise)
+      case Success(_) =>
+        logger.debug(s"{} Spark Driver is now in Unknown state as polling returned success", tag)
```

Comment:
Is a success that returns a driver state you're unaware of really a success or should it be an error? I can see arguments for either way but wanted to see what argument you went with.